cmd/parser/parse.go

```diff
@@ -0,0 +1,356 @@
+	cobraCmd := &cobra.Command{
+		Use:     "parse",
+		Aliases: []string{"p"},
+		Short:   "Parses a HAP rule entry validating its format",
+		Long:    "Parses a HAP rule entry validating its format, by default using simple string splitting. Documentation can be found ... .",
+		Example: `
+	# Parse a rule entry using simple string splitting
+	hap parse -e 'azure(PR=westeurope), aws->EU' 
+	
+	# Parse multiple rules from a file using simple string splitting
+	hap parse -e 'azure(PR=westeurope); aws->EU' 
+	
+	# Parse a rule entry using antlr parser and lexer
+	hap parse -g -e 'azure(PR=westeurope)' 
+
+	# Parse multiple rules from a file:
+	# --- rules.yaml
+	# rule:
+	# - azure(PR=westeurope)
+	# - aws->EU 
+	# ---
+	hap parse -f rules.yaml
+
+	# Sort rule entries by their priority
+	hap parse -p -e 'azure(PR=westeurope), aws->EU'	
+	
+	# Disable duplicated rule entries
+	hap parse -u -e 'azure(PR=westeurope), azure(PR=westeurope)'
+
+	# Check what rule will be matched and triggered against the provided test data
+	hap parse -p -u  -f ./correct-rules.yaml -m '{"plan": "aws"}'
+		`,
+
+		RunE:    func(_ *cobra.Command, args []string) error { 
+			return cmd.Run() 
+		},
+	}
+	cmd.cobraCmd = cobraCmd
+
+	cobraCmd.Flags().StringVarP(&cmd.rule, "entry", "e", "", "A rule to validate where each rule entry is separated by comma.")
+	cobraCmd.Flags().StringVarP(&cmd.match, "match", "m", "", "Check what rule will be matched and triggered against the provided test data. Only valid entries are taking into account when matching. Data is passed in json format, example: '{\"plan\": \"aws\", \"platformRegion\": \"cf-eu11\"}'.")
+	cobraCmd.Flags().StringVarP(&cmd.ruleFilePath, "file", "f", "", "Read rules from a file pointed to by parameter value. The file must contain a valid yaml list, where each rule entry starts with '-' and is placed in its own line.")
+	cobraCmd.Flags().BoolVarP(&cmd.useGrammar, "grammar", "g", false, "Use c parser and lexer generated with antlr instead of simple string splitting.")
+	cobraCmd.Flags().BoolVarP(&cmd.sort, "priority", "p", false, "Sort rule entries by their priority, in descending priority order.")
+	cobraCmd.Flags().BoolVarP(&cmd.unique, "unique", "u", false, "Display only non duplicated rules. Error entries are not considered for uniqueness.")
+	cobraCmd.Flags().BoolVarP(&cmd.signature, "signature", "s", false, "Mark rules with the mirrored signatures as duplicated. For example aws(PR=*, HR=westeurope) and aws(PR=westeurope, HR=*) are considered duplicated because of having mirrored signatures.")
+	cobraCmd.Flags().BoolVarP(&cmd.noColor, "no-color", "n", false, "Disable use color characters when generating output.")
+	cobraCmd.MarkFlagsOneRequired("entry", "file")
+
+	return cobraCmd
+}
+
+func (cmd *ParseCommand) Run() error {
+	cmd.parser = &rules.SimpleParser{}
+
+	if cmd.noColor {
+		colorError = ""
+		colorOk = ""
+		colorNeutral = ""
+		colorMatched = ""
+	}
+
+	if cmd.match != "" && (!cmd.sort || !cmd.unique) {
```

Comment:
I wonder if we need so many switches. 
Why not default mode running all necessary checks?
Or switching off particular tests if needed.